CMakeLists.txt

```diff
@@ -33,49 +50,86 @@ CPMAddPackage(
   OPTIONS "FMT_INSTALL YES" # create an installable target
 )
 
+# ---- Setup default project_options ----
+
+include(DynamicProjectOptions)
+# Initialize project_options variable related to this project
+# This overwrites `project_options` and sets `project_warnings`
+# uncomment the options to enable them:
+# cmake-format: off
+dynamic_project_options(
```

Comment:
I prefer to use `project_options` instead of `dynamic_project_options`. See the readme:
https://github.com/aminya/project_options

Reply:
Agreed, for **Ci**!
But for development in directory `all` not. I like the flexibility.

Another option would be a set of useful `cmake presets`.